### ISSUE TITLE
[sparse] BCOO mul fails for matrices with padding

### ISSUE THREAD

**Issue by jakevdp** [google/jax#8889]
See report in #8888

**Comment by jakevdp** [comment]
Turns out padding is not the issue, but perhaps duplicate indices are:
```python
import jax.numpy as jnp
from jax.experimental import sparse

indices = jnp.array([[0, 1, 0, 0, 1, 1],
                     [1, 0, 1, 2, 0, 2]]).T
data = jnp.array([1, 2, 3, 4, 5, 6])
m = sparse.BCOO((data, indices), shape=(3, 3))
print((m * m).todense())
# [[ 0 16 16]
#  [49  0  0]
#  [ 0  0  0]]
print(m.todense() * m.todense())
# [[ 0 16 16]
#  [49  0 36]
#  [ 0  0  0]]
```

**Comment by temirovazat** [comment]
import jax.numpy as jnp
from jax.experimental import sparse

indices = jnp.array([[0, 1, 0]]).T
data = jnp.array([1, 2, 3])
m = sparse.BCOO((data, indices), shape=(2,))
print((m * m).todense())
# [4 0]]
print(m.todense() * m.todense())
# [16  4]
